zipline/finance/blotter/blotter.py

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright 2018 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from zipline.finance.cancel_policy import NeverCancel
+from zipline.assets import Asset
+from zipline.utils.input_validation import expect_types
+
+
+class Blotter(object):
+
+    def __init__(self, data_frequency, cancel_policy=None):
+
+        self.data_frequency = data_frequency
+
+        self.cancel_policy = cancel_policy if cancel_policy else NeverCancel()
+
+        self.current_dt = None
+
+    def __repr__(self):
+        raise NotImplementedError
+
+    def set_date(self, dt):
+        self.current_dt = dt
+
+    @expect_types(asset=Asset)
+    def order(self, asset, amount, style, order_id=None):
+        """Place an order.
+
+        Parameters
+        ----------
+        asset : zipline.assets.Asset
+            The asset that this order is for.
+        amount : int
+            The amount of shares to order. If ``amount`` is positive, this is
+            the number of shares to buy or cover. If ``amount`` is negative,
+            this is the number of shares to sell or short.
+        style : zipline.finance.execution.ExecutionStyle
+            The execution style for the order.
+        order_id : str, optional
+            The unique identifier for this order.
+
+        Returns
+        -------
+        order_id : str or None
+            The unique identifier for this order, or None if no order was
+            placed.
+
+        Notes
+        -----
+        amount > 0 :: Buy/Cover
+        amount < 0 :: Sell/Short
+        Market order:    order(asset, amount)
+        Limit order:     order(asset, amount, style=LimitOrder(limit_price))
+        Stop order:      order(asset, amount, style=StopOrder(stop_price))
+        StopLimit order: order(asset, amount, style=StopLimitOrder(limit_price,
+                               stop_price))
+        """
+
+        raise NotImplementedError
```

Comment:
it is best practice to raise an instance of a class instead of the class itself, for example: `NotImplementedError()` vs `NotImplmentedError`. I also prefer mentioning which method was not implemented like:
```python
raise NotImplemented('order')
```

Reply:
Changed that